app/scripts/lib/fxa-client.js

```diff
@@ -110,7 +110,15 @@ function (_, FxaClient, $, xhr, p, Session, AuthErrors, Constants) {
 
       return self._getClient()
         .then(function (client) {
-          return client.signIn(email, password, { keys: shouldFetchKeys(relier) });
+          var signInOptions = {
+            keys: shouldFetchKeys(relier)
+          };
+
+          if (relier.has('service')) {
```

Comment:
This will result in multiple emails being sent. Both views/change_password and views/complete_reset_password call fxa-client->signIn, and both already cause an email to be sent. Instead of fetching the service from the relier, perhaps only the sign_in view should pass in the service.

Or, perhaps the call to signIn should be removed from the views and put into fxa-client.js->completePasswordReset and changePassword.

ATM, I'm somewhat partial to the 2nd. I feel like too much low-level account management stuff is leaking into the views, which should only be responsible for displaying & handling user-supplied input.